tasks/hooks/useOrganization.tsx

```diff
@@ -0,0 +1,89 @@
+import type { OrganizationDTO } from '@/mutations/organization_mutations'
+import { useOrganizationsForUserQuery } from '@/mutations/organization_mutations'
+import { OrganizationSwitchModal } from '@/components/OrganizationSwitchModal'
+import { useAuth } from '@/hooks/useAuth'
+
+export const LOCALSTORAGE_ORGANIZATION_KEY = 'organization'
+
+export type OrganizationContextValue = {
+  organization: OrganizationDTO | undefined,
+  setOrganization: Dispatch<SetStateAction<OrganizationDTO | undefined>>,
+  setOrganizationId: (organizationId: string) => void
+}
+
+export const OrganizationContext = createContext<OrganizationContextValue>({
+  organization: undefined,
+  setOrganization: () => {},
+  setOrganizationId: () => {},
+})
+
+export const useOrganization = () => useContext(OrganizationContext)
+
+/**
+ * TODO: Not a big fan of this useState() passing
+ * but I need to use the current organization AND applying ProvideOrganization in the same component
+ * See PageWithHeader.tsx
+ * @MaxSchaefer
+ */
+type ProvideOrganizationProps = {
+  isOrganizationSwitchModalOpen: boolean,
+  setOrganizationSwitchModalOpen: Dispatch<SetStateAction<boolean>>,
+  organization: OrganizationDTO | undefined,
+  setOrganization: Dispatch<SetStateAction<OrganizationDTO | undefined>>
+}
+
+export const ProvideOrganization = ({
+  children,
+  organization,
+  setOrganization,
+  isOrganizationSwitchModalOpen,
+  setOrganizationSwitchModalOpen
+}: PropsWithChildren<ProvideOrganizationProps>) => {
+  useAuth() // Calling useAuth() to prepare the context for later operations
+
+  const { data: organizations } = useOrganizationsForUserQuery()
+  const [storedOrganization, setStoredOrganization] = useLocalStorage<OrganizationDTO | undefined>(LOCALSTORAGE_ORGANIZATION_KEY, undefined)
+  // TOOD: See #26
+  // const [organization, setOrganization] = useState<OrganizationDTO>()
+  // const [isOrganizationSwitchModalOpen, setOrganizationSwitchModalOpen] = useState(false)
+
+  useEffect(() => {
+    if (storedOrganization) {
+      setOrganization(storedOrganization)
+    }
+  }, [])
+
+  useEffect(() => {
+    setOrganizationSwitchModalOpen(!organization)
+    if (organization) {
+      setStoredOrganization(organization)
+    }
+  }, [organization])
+
+  const setOrganizationId = (organizationId: string) => {
+    const organization = organizations?.find((organization) => organization.id === organizationId)
+    if (organization) {
+      setOrganization(organization)
+    }
+  }
+
+  return (
+    <OrganizationContext.Provider value={{ organization, setOrganization, setOrganizationId }}>
+      <OrganizationSwitchModal
+        id="organization-switch-modal"
+        isOpen={isOrganizationSwitchModalOpen}
+        organizations={organizations}
+        onDone={(organization) => {
+          setOrganization(organization)
+
+          // A manual switch of the organization will result in a page reload to refetch all the queries
+          window.location.reload()
```

Comment:
@JannikWibker the `window.location.reload()` is far from okay. I've tried to use `queryClient.clear()` and `queryClient.refechQueries()`, no effects. Can you take a look?